### PR TITLE
Add to do list

### PR DESCRIPTION
**Pull request by karam084**
-*  Project requirements

- Set up a new project with webpack that is based on the webpack exercise you have already completed.
- Create an index.html file and write your HTML markup here. Create an empty To Do List placeholder (<div> or <ul> element). The index.html file must be set as a template using the HTML Webpack Plugin.
- Create an index.js file and set an array of some simple to do tasks (array of objects). Each task object should contain three keys:
- description [string].
- completed [bool].
- index: [number].
- Write a function to iterate over the tasks array and populate an HTML list item element for each task.
- On page load render the dynamically created list of tasks in the dedicated placeholder. The list should appear in order of the index values for each task.
- Create a style.css and set rules for the To Do List. CSS must be loaded by Webpack Style/CSS Loader. Your list should be a clone of the part of the minimalist project captured in the video below.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,43 +4,16 @@
 
 To Do List project
```
Comment (medaminedev66) [inline README.md:5]: - [x] Kindly add a description of your project to the Readme file so users can know about your project.

File: README.md [hunk 2]
```diff
@@ -4,43 +4,16 @@
...
 - html and css
 - javascript
 - Visual studio and github
+- Webpack
 
-## Live Demo
+## Source code
 
-[Live Demo Link](https://karam084.github.io/online-courses-capstone/)
+[Demo](https://github.com/karam084/to_do_list)
```
Comment (medaminedev66) [inline README.md]: - [x] The demo link is leading to a wrong page. Kindly fix this.

File: README.md [hunk 3]
```diff
@@ -61,7 +34,8 @@ Give a ⭐️ if you like this project!
 ## Acknowledgments
 
 - Thank you Microverse for your effort.
-- Original design idea by https://www.behance.net/adagio07
+- Original design idea by https://web.archive.org/web/20180320194056/http://www.getminimalist.com:80/
+- https://www.youtube.com/watch?v=AcUd-_Yjjqg
 
 ## 📝 License
```
Comment (medaminedev66) [inline README.md:45]: - [x] The MIT license link is broken, kindly fix this. Your README is the entry point of your users and having it well written without any broken links gives a good first impression and maximizes the potential of your product to your users.

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      preload: false,
+      filename: 'demo.html',
+    }),
+    new ResourceHintWebpackPlugin(),
+  ],
+
+  stats: {
+    children: true,
+  },
+};
```
Comment (medaminedev66) [inline webpack.config.js]: - [x] The webpack is not configured well. It shows this error after running `npm start`:
image


- Kindly make sure that your project opened successfully after running `npm start`

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+const addButton = document.getElementsByTagName('button')[0];
+const incompleteTaskHolder = document.getElementById('incomplete-tasks');
+const completedTasksHolder = document.getElementById('completed-tasks');
+const createNewTaskElement = function tasks(taskString) {
+  const listItem = document.createElement('li');
+  const checkBox = document.createElement('input');
+  const label = document.createElement('label');
+  const editInput = document.createElement('input');
+  const editButton = document.createElement('button');
+
```
Comment (medaminedev66) [inline src/index.js:19]: - [ ] Following the project requirements, your project should have an array of some simple to-do tasks (array of objects) with each object containing three keys - description [string], completed [bool], index: [number]. You can create more properties or name these properties differently. Checkout the project requirement bellow 👇 :
image


- [x] Your project also should have a function that iterates over the tasks array and populates the HTML list item element for each task.

File: README.md [hunk 6]
```diff
@@ -61,7 +34,8 @@ Give a ⭐️ if you like this project!
 ## Acknowledgments
 
 - Thank you Microverse for your effort.
-- Original design idea by https://www.behance.net/adagio07
+- Original design idea by https://web.archive.org/web/20180320194056/http://www.getminimalist.com:80/
+- https://www.youtube.com/watch?v=AcUd-_Yjjqg
```
Comment (medaminedev66) [inline README.md]: - [x] These links doesn't have any relation with your project. Kindly remove them to make your project professional.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @karam084 , :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
```
Comment (mmsesay) [inline webpack.config.js]: - [x] I see you have worked on the required changes. However, it seems like the npm problem is still around. See below screenshots of the errors I encounter upon running `npm start`
Please try to check this out.

Shot 1
a1

Shot 2
a2

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
+import './style.css';
+import './style.scss';
+
+const taskInput = document.getElementById('new-task');
+const addButton = document.getElementsByTagName('button')[0];
+const incompleteTaskHolder = document.getElementById('incomplete-tasks');
+const completedTasksHolder = document.getElementById('completed-tasks');
+const createNewTaskElement = function tasks(taskString) {
```
Comment (mmsesay) [inline src/index.js:13]: - [ ] As suggested by the previous reviewer, you should create an array with to-do tasks with each object containing three keys.
See example:
```
const tasks = [
  {
    description: 'An example description in string',
    completed: false,
    index: 10,
  },
];
```

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @karam084, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @karam084 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- [ ] There should be no custom changes to linter config. Kindly set up the linter again in your repository.
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Amrhub — APPROVED** [review]
# Approved 💯 ✅ 🥳 🎈 🎉 

Hi @karam084, 👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights ✅ 
Highly organized and clean code ✔️ 👨🏻‍💻 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**